### ISSUE TITLE
Rework NSS port as an UNIX variant

### ISSUE THREAD

**Issue by kenz-gelsoft** [kenz-gelsoft/gecko-dev#23]
from https://github.com/kenz-gelsoft/gecko-dev/issues/16#issue-2451675209

will replace or reduce https://github.com/kenz-gelsoft/gecko-dev/pull/17

**Comment by kenz-gelsoft** [comment]
It seems korli did this work kindly.

- nspr
  - https://github.com/haikuports/haikuports/commit/0172f2a465542c1e03e5b5e89dcf4094bf6d6984
- nss
  - https://github.com/haikuports/haikuports/commit/f4b16f6d35e7b485e653e9191f4702a6769f73f3

**Comment by kenz-gelsoft** [comment]
We should be able to revert these commits now 1b17ce5299c53a99f59e78f5ce15dc8a85ca7b31...534ed235dac9e309750c836b820692f5821fc2b2

**Comment by kenz-gelsoft** [comment]
was not so simple
```
79:00.90 toolkit/library/build/libxul.so
79:24.14 ld.lld: error: undefined symbol: arc4random_buf
79:24.14 >>> referenced by std.9b903c7c25c6c255-cgu.10
79:24.14 >>>               std-2d470cba23f40e31.std.9b903c7c25c6c255-cgu.10.rcgu.o:(std::sys::random::hashmap_random_keys::h6c04576eeb078751) in archive ../../../x86_64-unknown-haiku/debug/libgkrust.a
79:24.70 clang++: error: linker command failed with exit code 1 (use -v to see invocation)
79:24.70 /boot/home/src/gecko-dev2/config/rules.mk:538: recipe for target '../../../dist/bin/libxul.so' failed
79:24.70 make[4]: *** [../../../dist/bin/libxul.so] Error 1
79:24.70 /boot/home/src/gecko-dev2/config/recurse.mk:72: recipe for target 'toolkit/library/build/target' failed
79:24.70 make[3]: *** [toolkit/library/build/target] Error 2
79:24.70 /boot/home/src/gecko-dev2/config/recurse.mk:32: recipe for target 'compile' failed
79:24.70 make[2]: *** [compile] Error 2
79:24.70 /boot/home/src/gecko-dev2/config/rules.mk:361: recipe for target 'default' failed
79:24.70 make[1]: *** [default] Error 2
79:24.71 client.mk:59: recipe for target 'build' failed
79:24.71 make: *** [build] Error 2
79:24.81 W 140 compiler warnings present.
79:25.60 W Notification center failed: Install notify-send (usually part of the libnotify package) to get a notification when the build finishes.
~/src/gecko-dev2> 
```

**Comment by kenz-gelsoft** [comment]
```
~/src/haiku> grep -R arc4random_buf . --exclude-dir=generated/ --exclude-dir=.git
./headers/compatibility/bsd/stdlib.h:void               arc4random_buf(void *buf, size_t nbytes);
./src/bin/network/ping/ping6.c: arc4random_buf(nonce, sizeof(nonce));
./src/libs/bsd/arc4random.c:arc4random_buf(void *buf, size_t n)
./src/libs/compat/freebsd_network/compat/sys/libkern.h:void arc4random_buf(void *ptr, size_t len);
./src/libs/compat/freebsd_network/libkern.cpp:arc4random_buf(void *ptr, size_t len)
./src/libs/compat/openbsd_wlan/net80211/ieee80211_node.c:               arc4random_buf(ic->ic_globalcnt, EAPOL_KEY_NONCE_LEN);
./src/libs/compat/openbsd_wlan/net80211/ieee80211_node.c:               arc4random_buf(k->k_key, k->k_len);
./src/libs/compat/openbsd_wlan/net80211/ieee80211_node.c:                       arc4random_buf(k->k_key, k->k_len);
./src/libs/compat/openbsd_wlan/net80211/ieee80211_node.c:       arc4random_buf(ni->ni_nonce, EAPOL_KEY_NONCE_LEN);
./src/libs/compat/openbsd_wlan/net80211/ieee80211_pae_input.c:  arc4random_buf(ic->ic_nonce, EAPOL_KEY_NONCE_LEN);
./src/libs/compat/openbsd_wlan/net80211/ieee80211_proto.c:      arc4random_buf(k->k_key, k->k_len);
./src/libs/compat/openbsd_wlan/net80211/ieee80211_proto.c:              arc4random_buf(k->k_key, k->k_len);
./src/libs/libtelnet/pk.c:      arc4random_buf(seed, seedsize);
```

**Comment by kenz-gelsoft** [comment]
Maybe adding `-lnetwork` or `-lbsd` somewhere will fix this, but I'll handle later.

**Comment by kenz-gelsoft** [comment]
Fixed by  #87
